src/AudioUnitSDK/AUEffectBase.cpp

```diff
@@ -193,7 +193,7 @@ void AUEffectBase::MaintainKernels()
 
 	if (mKernelList.size() < nKernels) {
 		mKernelList.reserve(nKernels);
-		for (auto i = static_cast<UInt32>(mKernelList.size()); i < nKernels; ++i) {
+		for (auto i = mKernelList.size(); i < nKernels; ++i) {
```

Comment:
all this achieves is making comparison between `i` and `nKernels` a type width mismatch (`UInt32` vs `size_t`)?

Reply:
Not exactly: it is only a problem if nkernels was size_t and i was not. This behavior is well defined within the C++ and C standards, but I could change it back. I just wanted to make the code look cleaner.

Reply:
`nKernels` is `UInt32` and, with your proposed change, `i` is `size_t` (which is 64-bit on all of our supported architectures). Definitely yes this is still well-defined, but it will introduce a compiler warning with certain settings, and the current version does not.